packages/taiko-client/preconfapi/builder/blob.go

```diff
@@ -34,40 +35,79 @@ func NewBlobTransactionBuilder(
 // return an unsigned transaction, intended for preconfirmations.
 func (b *BlobTransactionBuilder) BuildUnsigned(
 	_ context.Context,
-	txListBytes []byte,
-	l1StateBlockNumber uint32,
-	timestamp uint64,
-	coinbase common.Address,
-	extraData [32]byte,
+	opts BuildUnsignedOpts,
 ) (*types.Transaction, error) {
-	compressedTxListBytes, err := utils.Compress(txListBytes)
-	if err != nil {
-		return nil, err
-	}
+	encodedParams := make([][]byte, 0)
 
-	var blob = &eth.Blob{}
-	if err := blob.FromData(compressedTxListBytes); err != nil {
-		return nil, err
+	blobs := make([]*eth.Blob, 0)
+
+	var offset uint64
+
+	var totalTxBytes []byte
+
+	for i, opt := range opts.BlockOpts {
+		txListBytes, err := signedTransactionsToTxListBytes(opt.SignedTransactions)
+		if err != nil {
+			return nil, err
+		}
+
+		compressedTxListBytes, err := utils.Compress(txListBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		totalTxBytes = append(totalTxBytes, compressedTxListBytes...)
+
+		params := &encoding.BlockParams{
+			Coinbase:           common.HexToAddress(opt.Coinbase),
+			ExtraData:          rpc.StringToBytes32(opt.ExtraData),
+			Signature:          []byte{}, // no longer checked
+			L1StateBlockNumber: opt.L1StateBlockNumber,
+			Timestamp:          opt.Timestamp,
+		}
+
+		if opts.MultipleBlobs {
```

Comment:
Is there a reason to have this option instead of always trying to pack as many tx lists in a single blob? Maybe easier code for now so for now not trying to be smart?

Reply:
yeah it's just a lazy way for MVPs for testnet right now, next iteration can just check if the max bytes for a blob has been exceeded or not

Reply:
pushed up a change

Reply:
I would strongly prefer not to merge this feature into our main branch.

Reply:
Why?

Reply:
> I would strongly prefer not to merge this feature into our main branch.

any reason why? it was an ask from a preconf team, but I can tell them no if we have a good reason.

Reply:
If in one Ethereum transaction,  the same blob is used for more than one L2 block, then the proposer shall simply propose one larger block instead of two to avoid unnecessary cost overhead. If these two blocks are not sharing the same blob, I'd prefer each block is proposer separately using 'proposeBlock' to avoid chaning 'proposeblock' to 'proposeBlocks' (or another wrapper function can be added either in the contract or in a wrapper contract).

Reply:
> If in one Ethereum transaction, the same blob is used for more than one L2 block, then the proposer shall simply propose one larger block instead of two to avoid unnecessary cost overhead. If these two blocks are not sharing the same blob, I'd prefer each block is proposer separately using 'proposeBlock' to avoid chaning 'proposeblock' to 'proposeBlocks' (or another wrapper function can be added either in the contract or in a wrapper contract).

OK, so you are fine with adding "proposeBlocks" function separately, and to remove logic of multiple blocks in one blob.

Reply:
Yes, as long as `proposeBlocks` simply calls '_proposeBlock' multiple times without different logics :D

Reply:
Multiple blocks inside a single blob is also a feature request of nethermind. Their preconfs would commit to the tx trie root which is only available in a block header, and they will do so every 3 seconds. I think in theory this does not necessarily require multiple blocks, but it certainly makes checking the preconf condition very easy (check the value in the block header), they will be creating multiple L2 blocks per L1 block in any case. Blob data slices also makes it possible to share blob data with other based rollups.

The overhead of multiple blocks can also be avoided, though that it is a big change so unlikely to be done for the current protocol version.